joomla/joomla-cms
# Fixing missing support for "Subcategory Levels" parameter in contacts category view

This adds the support for `Subcategory Levels`  in `Category` tab of  `List Contacts in a Category` view. Currently, this setting is not applied in the query, so using it will not allow viewing all contacts in the category and its subcategories.

### Summary of Changes
Added state entry and query filter for this parameter.


### Testing Instructions
- Create a contact category called `All contacts`
- Create a contact category called `A` and set the parent as `All contacts`
- Create a contact category called `B` and set the parent as `All contacts`
- Create contact `Contact from A` and place it in `A` category
- Create contact `Contact from B` and place it in `B` category
- Create a menu item of type `List Contacts in a Category`, chose category `All contacts` and make sure in `Category` tab `Subcategory Levels` is set to `All` or any positive value
- Save and head to the recently created menu item in front-end


### Actual result BEFORE applying this Pull Request
There should be no contacts displayed in this view.


### Expected result AFTER applying this Pull Request
All contacts from category `A` and `B` will be displayed.


### Link to documentations
Please select:
